prow/config/jobs/istio.yaml

```diff
@@ -402,6 +402,12 @@ jobs:
     regex: '^samples/bookinfo/src/'
     requirements: [docker]
 
+  # Try to create the Mac binaries. Ocassionally a change breaks the ability to build on MacOS.
+  - name: macbuildcheck
+    types: [presubmit]
+    modifiers: [presubmit_optional]
+    command: [TARGET_OS=darwin, make, build]
```

Comment:
would this work? perhaps `[make, -e, "TARGET_OS=darwin", build]` ?

also, does it make sense to restrict this job to changes to Makefiles?

Reply:
> also, does it make sense to restrict this job to changes to Makefiles?

No, as changes in code typically cause this to happen. Especially changes in one of the Linux only files which may require a change in the MacOS only file for compatibility.

Reply:
`make -e TARGET_OS=darwin build` does also seem to build the files if run locally, as well as the prior command. I suspect this might be better since the common-files script could maybe not pass TARGET_OS in the future.

Reply:
Sorry, my suggestion above was just to rearrange the order of the arguments, as the final yaml will be rendered having `TARGET_OS=darwin` as the first item in the array, not sure if it would work, as it's not a binary.